Automattic/sensei
# When tour is closed or skipped, persist the choice and don't load that tour for that user

Resolves https://github.com/Automattic/sensei/issues/7501

## Proposed Changes
When a user closes or skips a tour, we're presenting that data here so that the user does not see that specific tour again.

## Testing Instructions


1. Run `npm` and `composer` installation
2. Go to Course editor by creating a new Course
3. Make sure the the Tour does not appear until you complete/close the Course creation modals
4. Make sure you see a tour loaded
5. Without doing anything else, just refresh the page
6. Make sure the tour still loads
7. Now click on the skip button or the Cross button on that tour modal
8. Refresh the page
9. Make sure you don't see the Course tour loading again
10. Now go to a lesson
11. Make sure you see the lesson tour. Seeing the lesson tour confirms that Persisting status for one tour isn't affecting another tour
12. Now skip or Cross the lesson tour
13. Now refresh the page
14. Make sure you don't see the lesson tour
15. Go back to Course edit page again
16. Make sure you don't see the tour

If you want to retest, open database table `wp_usermeta` and delete the rows with key `sensei_tours`. The tour will appear again.

https://github.com/Automattic/sensei/assets/6820724/0e5434b3-58ad-4681-bd91-94efd703d2b9

## New/Updated Hooks


* `sensei_tour_is_complete` - Allows user to override the status of a tour being complete, to either load or not load it.

## Pre-Merge Checklist
